tests/components/smappee/test_config_flow.py

```diff
@@ -10,7 +19,181 @@
+            CONF_HOSTNAME: "Smappee1006000212.local.",
+            "type": "_ssh._tcp.local.",
+            "name": "Smappee1006000212._ssh._tcp.local.",
+            "properties": {"_raw": {}},
+        },
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "connection_error"
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_connection_error(hass):
+    """Test we show user form on Smappee connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data={"host": "1.2.3.4"},
+    )
+
+    assert result["reason"] == "connection_error"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_connection_error_empty_host(hass):
+    """Test we show user form on Smappee connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data={"host": None},
+    )
+
+    assert result["reason"] == "connection_error"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_zeroconf_no_data(hass):
+    """Test we abort if zeroconf provides no data."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}
+    )
+
+    assert result["reason"] == "connection_error"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_zerconf_wrong_mdns(hass):
+    """Test we abort if unsupported mDNS name is discovered."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data={
+            "host": "1.2.3.4",
+            "port": 22,
+            CONF_HOSTNAME: "example.local.",
+            "type": "_ssh._tcp.local.",
+            "name": "example._ssh._tcp.local.",
+            "properties": {"_raw": {}},
+        },
+    )
+
+    assert result["reason"] == "invalid_mdns"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_user_device_exists_abort(hass):
+    """Test we abort user flow if Smappee device already configured."""
+    with patch("pysmappee.api.SmappeeLocalApi.logon", return_value={}), patch(
+        "pysmappee.api.SmappeeLocalApi.load_advanced_config",
+        return_value=[{"key": "mdnsHostName", "value": "Smappee1006000212"}],
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_command_control_config", return_value=[]
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_instantaneous",
+        return_value=[{"key": "phase0ActivePower", "value": 0}],
+    ):
+        result = await hass.config_entries.flow.async_init(
```

Comment:
Add a mock config entry to home assistant instead for the first entry.

https://github.com/home-assistant/core/blob/ad22619efb5de2409216ec4aff5e9f8801f5b3cf/tests/common.py#L652